app/components/shared/inputs/TextInput.tsx

```diff
@@ -0,0 +1,66 @@
+import { Component, Prop } from 'vue-property-decorator';
+import cx from 'classnames';
+import styles from './TextInput.m.less';
+import { BaseInput } from './BaseInput';
+import { ITextMetadata } from './index';
+import { $t } from 'services/i18n';
+
+@Component({})
+export default class TextInput extends BaseInput<string, ITextMetadata> {
+  @Prop() readonly value: string;
+
+  @Prop({ default: () => ({}) })
```

Comment:
I'm a little confused as to what is going on here.  Why is the default value a function?

Reply:
i don't remember actually but that's not a change in the code. i think it's related to the metadata helper function or something like that

Reply:
Vue requires using a function for default values if the default value is Object

Reply:
Huh.  Interesting.